docs/server/upgrading.rst

```diff
@@ -3,170 +3,251 @@
+   Before upgrading we **strongly** recommend that you
+   :ref:`backup your current system <backup>`.
 
-  - **Carefully read this page** before proceeding
-  - Make all the recommended **backups**
-  - Try to **follow these instructions** as closely as possible
 
+.. _upgrading#db-migration:
 
-This page is divided in three sections:
+Migrate your database
+---------------------
 
-1. Preparatory tasks that should be performed before upgrading.
-2. Detailed steps to actually perform the upgrade.
-3. Suggested tasks to fine tune the setup after upgrading.
+If you are currently using SQLite for your database you will need to 
+:doc:`migrate to either MySQL or PostgreSQL <database_migration>`
+before you upgrade.
 
 
-.. _upgrading#preparatory-tasks:
+.. _upgrading#latest-changes:
 
-Preparatory tasks
------------------
+Latest changes
+--------------
 
-Before upgrading Pootle to a newer version, make sure to go through this
-checklist.
+Before upgrading Pootle familiarize yourself with :doc:`important changes
+</releases/index>` since the version that you are upgrading from.
 
-* Familiarize yourself with :doc:`important changes </releases/index>` in
-  Pootle.
 
-* If you want to change the database backend then have a look at the
-  :doc:`database migration <database_migration>` page first. We discourage
-  using SQLite, so if you are using it please migrate to a real database
-  server.
+.. _upgrading#requirements:
 
-* Ensure that you meet the :ref:`hardware requirements
-  <installation#hardware_requirements>` for the newer version.
+System requirements
+-------------------
 
-* Always make backups of all your translation files (your whole
-  :setting:`POOTLE_TRANSLATION_DIRECTORY`). Use the :djadmin:`sync_stores`
-  command to synchronize all your translation files to disk before making any
-  backup.
+You should check that you have all of the necessary :ref:`system requirements
+<requirements>` and have installed all required :ref:`system packages
+<requirements#packages>`.
 
-* Also backup your settings, to avoid losing any settings customizations.
+.. warning::
 
-* Make a backup of your complete database using the appropriate *dump*
-  command for your database system. For example :command:`mysqldump` for MySQL,
-  or :command:`pg_dump` for PostgreSQL.
+   Pootle 2.7.0 or newer requires **Python 2.7**
 
-* And don't forget to backup any code, templates or styling customization that
-  you have done to your installation.
+   If you are upgrading from a virtual environment using an earlier Python
+   version, you **must** upgrade or rebuild your virtual environment first.
 
-* Familiarize yourself with any new :ref:`settings <settings#available>` that
-  have been introduced.
 
+.. _upgrading#update-pip:
 
-.. _upgrading#upgrading:
+Update pip
+----------
 
-Upgrading
----------
+You should now upgrade Pip to the latest version:
+
+.. code-block:: bash
 
-Upgrading Pootle using the :command:`pip`.
+   (env) $ pip install --upgrade pip
```

Comment:
This is the first command displayed on the upgrade instructions and it uses a virtualenv, but the instructions don't tell to enable the virtualenv before running this. IMHO it is better to state clearly that the virtualenv must be enabled before.